cli/Sources/TuistCAS/Services/KeyValueService.swift

```diff
@@ -94,11 +98,26 @@ public struct KeyValueService: CompilationCacheService_Keyvalue_V1_KeyValueDB.Si
                 serverURL: serverURL
             ) {
                 var value = CompilationCacheService_Keyvalue_V1_Value()
+
+                // Store entry keys for cache insights
+                var entryKeys: [String] = []
+
                 for entry in json.entries {
                     if let data = Data(base64Encoded: entry.value) {
                         value.entries["value"] = data
+                        entryKeys.append(entry.value)
+                    }
+                }
+
+                // Save the entry keys to a JSON file asynchronously (non-blocking)
```

Comment:
Will this mean there's a chance of missing some? (e.g. if on CI the server is terminated before entries are persisted)

Reply:
> Will this mean there's a chance of missing some? (e.g. if on CI the server is terminated before entries are persisted)

I'd expect on the CI most misses to lead to uploads. But yes, this is a best-effort basis and we _might_ mislabel some remote hits for misses. Unfortunately, couldn't figure out a better way.